src/components/PersonItem/PersonItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { PeopleContex } from '../../peopleContext';
+
+type Props = {
+  person: Person;
+};
+
+const getPersonByName = (
+  name: string | null,
+  people: Person[],
+): Person | undefined => {
+  return people.find(person => person.name === name);
+};
```

Comment:
The `getPersonByName` function is a utility function that could be used in other components or parts of the application. Consider moving it to a separate utility file.